docs/README.md

```diff
@@ -0,0 +1,86 @@
+# Determined docs development
```

Comment:
# Documentation Guide

## Writing Documentation

Docs are generated using Sphinx. Documentation is written in reStructuredText, or in MyST Markdown.

The documentation build process is stored in the `Makefile`.  Running `make live` will generate HTML formatted documentation.

Before submitting a pull request, run [rstfmt](https://pypi.org/project/rstfmt/).

## Using `make live`

When creating and editing files, you can use `make live` to simulate live docs
and automatically rebuild the docs whenever you save a change.

To try it out, run `make live` in a terminal and visit
http://localhost:1234 in your browser.

Typing Ctrl+C interrupts `make live` and stops the docs from rendering in
the browser. This way, you won't continue making edits and wonder
why your edits aren't appearing in your browser.

## Algolia Search

[Aloglia](https://www.algolia.com) search is a search-as-a-service provider.
Algolia indexes our site via their [crawler](https://crawler.algolia.com/admin),
then we include their search bar component into our website, so that searches in
the search bar obtain results based on Algolia's hosted search index for our
site.

We configure and inject the Algolia search bar into each page via
[this JavaScript](assets/scripts/docsearch.sbt.js).

### Relative Search Results

By default, Algolia search results return absolute URLs to
`https://docs.determined.ai`, based on the actual URL of searched pages.
The effect is that somebody searching the docs hosted on their
on-prem cluster would be redirected to `https://docs.determined.ai` when they
click on a search result.  This would be very annoying for users.

Fortunately, Algolia allows us to define a `transformItems` function that can
make arbitrary client-side changes to the search results before displaying them
to a user.  The relativization happens in the JS code, but the key feature
required to relativize the search results is to know the relative path from the
current page to the root of the docs.  We embed this into every docs page's
header as a special `rel=root` link inside
[one of our template overrides](_templates/page.html).

### Versioning

You can not only view the latest docs, but you can also view older doc versions. Likewise, the search results you see are for the version of docs you are viewing. 
To enable this, we have configured the Algolia crawler to be version aware. 

The Algolia crawler applies a version tag to all results. Algolia
simply extracts the information from the version component of the URL path as it
crawls.

Since the resulting search index does not automatically support filtering by 
version; we added the `version` tag to the `determined` Algolia index's
`attributesForFaceting` setting.  It is configured as a "filter-only" facet.

To actually filter by version, we need to know which version of the docs
the current page was built against.  This information is embedded in the same
`rel=root` link mentioned above.

### Sphinx-Native Search Fallback

We have some users whose clusters do not have internet access.  For those users
Algolia's search-as-a-service model will never work.  When we detect
that we are unable to load resources from Aloglia, we fall back to the default
Sphinx search.  See the JavaScript for implementation.

### Dev Builds

Since Algolia indexing happens after docs are published to docs.determined.ai,
development builds of docs cannot rely on a versioned Algolia index.  
Instead, dev builds search the `latest` Algolia index, which is based on the
most recent _published_ version of Determined.

This means dev builds of docs will never return expected results. Builds
of unreleased docs will have out-of-date search results, while builds of old
docs will have search results from the future.  Because the Sphinx search 
results are ineffectual, this is an acceptable tradeoff.

### Canonical URLs

Our site configures canonical links to point to `/latest` all the time.  This
is necessary for SEO optimization, and is also a common practice on other docs sites (python standard library docs, for instance).  As a result, the Algolia crawler must 
be configured with `ignoreCanonicalTo: true` before it will index anything other
than `/latest`.

## Theming

The html templates in the _templates directory are used to override the Sphinx Book theme.

@johnkim-det you can add more content here: